sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/GitScmProvider.java

```diff
@@ -47,4 +67,50 @@ public boolean supports(File baseDir) {
   public BlameCommand blameCommand() {
     return this.jgitBlameCommand;
   }
+
+  @Nullable
+  @Override
+  public Collection<Path> branchChangedFiles(String targetBranchName, Path rootBaseDir) {
+    RepositoryBuilder builder = new RepositoryBuilder().findGitDir(rootBaseDir.toFile());
+    if (builder.getGitDir() == null) {
+      LOG.warn("Not inside a Git work tree: {}", rootBaseDir);
```

Comment:
The existing blame is actually throwing an exception if this happens, so we should throw it here too. It also seems like you should use `.setMustExist(true);` otherwise it might create a repo. Even better, maybe create a factory to be used by both classes.